### PR TITLE
Add codec options to parquet writer

### PR DESCRIPTION
**Pull request by yaqi-zhao**
Velox have upgrade arrow to 14.0.0 which support CodecOptions. 
If we add CodecOptions to ::parquet::WriterOptions it will help to make more compression parameters (such as window_bits) customizable when creating the Codec for parquet writer.

### CODE REVIEW

**Review by pedroerp — APPROVED** [review]

File: velox/dwio/parquet/writer/Writer.h [hunk 1]
```diff
@@ -94,6 +94,7 @@ struct WriterOptions {
   // The default factory allows the writer to construct the default flush
   // policy with the configs in its ctor.
   std::function<std::unique_ptr<DefaultFlushPolicy>()> flushPolicyFactory;
+  ::arrow::util::CodecOptions codecOptions;
```
Comment (pedroerp) [inline velox/dwio/parquet/writer/Writer.h:97]: you're missing an include or something. Check the CI failure.

**Review by pedroerp — CHANGES_REQUESTED** [review]

File: velox/dwio/parquet/writer/Writer.h (hunk 1 above)
Comment (yaqi-zhao) [reply velox/dwio/parquet/writer/Writer.h]: I add the included header and run success on my machine

File: velox/dwio/parquet/writer/Writer.h [hunk 2]
```diff
@@ -25,8 +25,12 @@
 #include "velox/dwio/common/WriterFactory.h"
 #include "velox/vector/ComplexVector.h"
 
+#include "velox/dwio/parquet/writer/arrow/Properties.h"
```
Comment (pedroerp) [inline velox/dwio/parquet/writer/Writer.h:28]: nit: put the include in the same list above so that clang-format can order them properly

**Comment by yaqi-zhao** [comment]
Hi, @pedroerp . There is 1 warning in Facebook Internal - Linter. But I cannot see the details since I'm not a meta employee. Can you give me some hint how to solve the warning? Thanks!

**Comment by pedroerp** [comment]
> Hi, @pedroerp . There is 1 warning in Facebook Internal - Linter. But I cannot see the details since I'm not a meta employee. Can you give me some hint how to solve the warning? Thanks!

It was complaining about the missing include but it shows as cleared now. I'm retrying one internal CI job but will get it merged today. Thanks!

**Pull request closed without merging**